### PR TITLE
[deps] Migrate dependencie rest-auth to dj-rest-auth #168

### PR DESCRIPTION
**Pull request by tullur**
Django-rest-auth is not actively maintained, switched to the maintained fork dj-rest-auth.
upd: `dj-rest-auth` replaced `rest_framework_jwt` with `django-rest-framework-simplejwt` (jazzband/dj-rest-auth#3) and now 'RegisterView' object has two attributes 'access_token' and 'refresh_token'

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ passlib>=1.7.1,<1.8.0
 djangorestframework-link-header-pagination>=0.1.1,<0.2.0
 weasyprint>=43,<52
 openwisp-users~=0.4.0
-django-rest-auth>=0.9.3,<0.12.0
+dj-rest-auth ~= 1.1.2
```
Comment (nemesisdesign) [inline requirements.txt:8]: can you remove the spaces here please?

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Thanks @tullur, see my comment below.

File: requirements.txt (hunk 1 above)
Comment (tullur) [reply requirements.txt]: Yikes

Comment (tullur) [reply requirements.txt]: Removed

**Comment by tullur** [comment]
Test failed with:

`from rest_framework_simplejwt.serializers import TokenObtainPairSerializer`
`ModuleNotFoundError: No module named 'rest_framework_simplejwt'`

But if you add rest_framework_simplejwt module to the project and run tests, it would be failed:
 
`{'user': user, 'token': self.token}, context=context`
`AttributeError: 'RegisterView' object has no attribute 'token'`

**Comment by tullur** [comment]
Hey @nemesisdesign , I fixed problem with token, dj-rest-auth replaced `rest_framework_jwt` with `django-rest-framework-simplejwt` (https://github.com/jazzband/dj-rest-auth/pull/3) and  now 'RegisterView' object has two attributes 'access_token' and 'refresh_token'

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Can you rebase on the current master please?

Something like this:
```
git remote add upstream https://github.com/openwisp/openwisp-radius.git
git fetch upstream
git rebase upstream/master
```

Then, can you reformat the code with `openwisp-qa-format` please?

```
openwisp-qa-format
```

File: requirements.txt [hunk 2]
```diff
@@ -5,10 +5,11 @@ passlib>=1.7.1,<1.8.0
...
 openwisp-users~=0.4.0
-django-rest-auth>=0.9.3,<0.12.0
+dj-rest-auth~=1.1.2
 django-sendsms>=0.4.0,<0.5.0
 jsonfield~=3.1.0
 django-private-storage~=2.2
 celery~=4.4.0
 redis~=3.5.0
 django-ipware~=3.0.0
+djangorestframework-simplejwt~=4.4.0
```
[the remark this answers is not in the document]
Comment (tullur) [reply requirements.txt:15]: Yep

**Review by nemesisdesign — APPROVED** [review]
Thanks @tullur!
I will finish to improve some details in the `dev` branch.
I found out the definition of the JWT requirement in dj-rest-auth is not usable via pip so we will have to redefine it but we can do so in the test settings.

**Comment by nemesisdesign** [comment]
@tullur if you want to keep helping OpenWISP, please read [Help Us to Grow](https://openwisp.io/docs/general/help-us.html), it contains small pieces of advice which can help us a lot. Thank you.

**Comment by tullur** [comment]
@nemesisdesign Ok, thank you for help and reviewing.

**Pull request merged**